src/cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs

```diff
@@ -248,6 +248,25 @@ public static void CopyTo(this IBuffer source, uint sourceIndex, IBuffer destina
 
 #endregion (IBuffer).CopyTo extensions for copying to an (IBuffer)
 
+#region (IBuffer).TryGetBuffer extensions for getting the underlying managed array if backed by one
+
+        public static bool TryGetBuffer(this IBuffer source, out ArraySegment<byte> array)

```

Comment:
Per previous offline conversation, should we call this `TryGetArray` to be similar to the mentioned naming conversation in the comments and to signify the fact that calling `TryGetBuffer` on an `IBuffer` type might not be clear on what you are getting and `TryGetArray` might be a bit better to signify you get an array if it is allocated in managed memory.